### PR TITLE
View all racks 

### PR DESCRIPTION
**Pull request by clairecfu**
- [x] added racks/get-all endpoint to backend
- [x] added view all racks button to front end 

**NOTED: these will need to be updated with datacenter logic, have created tasks for these in notion and am tracking

### CODE REVIEW

File: rackcity/views/rack_views.py [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
-
+from rackcity.views.rackcity_utils import get_rack_detailed_response
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])
+def rack_get_all(request):
+    """
+    List all racks
+    """
+    racks = Rack.objects.all()
+    return get_rack_detailed_response(racks)
```
Comment (clairecfu) [inline rackcity/views/rack_views.py]: get-all endpoint

File: rackcity/views/rack_views.py [hunk 2]
```diff
@@ -27,32 +35,7 @@ def rack_get(request):
...
-            "rack": rack_serializer.data,
-            "instances": instances_serializer.data,
-        }
-        racks_with_instances.append(rack_detail)
-
-    return JsonResponse(
-        {"racks": racks_with_instances},
-        status=HTTPStatus.OK
-    )
+    return get_rack_detailed_response(racks)
```
Comment (clairecfu) [inline rackcity/views/rack_views.py]: refactored this into a shared util function

File: src/components/elementView/rackTab.tsx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+              minimal
+              intent={Intent.DANGER}
+              onClick={this.handleDeleteOpen}
+            />
+          </div>
+        ) : null}
+        <Button
+          text="View All Rack(s)"
+          onClick={(e: any) => this.getAllRacks()}
+        />
```
Comment (clairecfu) [inline src/components/elementView/rackTab.tsx]: adding the view all racks button

**Comment by clairecfu** [comment]
lots of other FE changes in this PR, these are from a diff branch (landing view refactor) that are still waiting to get merged

File: rackcity/views/rack_views.py [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
 from rackcity.models import Rack, ITInstance
-from rackcity.api.serializers import RackSerializer, ITInstanceSerializer, RecursiveITInstanceSerializer
+from rackcity.api.serializers import RackSerializer, ITInstanceSerializer
 from rackcity.api.objects import RackRangeSerializer
 from rest_framework.decorators import permission_classes, api_view
 from rest_framework.permissions import IsAuthenticated, IsAdminUser
 from http import HTTPStatus
-
+from rackcity.views.rackcity_utils import get_rack_detailed_response
+@api_view(['GET'])
```
Comment (juliasaveliff) [inline rackcity/views/rack_views.py]: this is incredibly picky but should add 2 empty lines here for linting 🤪

File: rackcity/views/rackcity_utils.py [hunk 5]
```diff
@@ -1,5 +1,37 @@
...
+from http import HTTPStatus
+from django.http import JsonResponse
+
+
+def get_rack_detailed_response(racks):
+    if racks.count() == 0:
+        return JsonResponse(
+            {"failure_message": "There are no existing racks within this range. "},
+            status=HTTPStatus.BAD_REQUEST,
+        )
```
Comment (juliasaveliff) [inline rackcity/views/rackcity_utils.py]: we can refactor this later when we dig into user-friendliness of error messages, but I don't think a user's attempt to view all should show the same "range" failure message if there are no racks

File: rackcity/views/rack_views.py (hunk 2 above)
Comment (juliasaveliff) [reply rackcity/views/rack_views.py]: b.e.a.u.t.i.f.u.l.

File: rackcity/views/rack_views.py (hunk 4 above)
Comment (clairecfu) [reply rackcity/views/rack_views.py]: we need AUTO LINTING

File: rackcity/views/rackcity_utils.py (hunk 5 above)
Comment (clairecfu) [reply rackcity/views/rackcity_utils.py]: sounds good - tracking this for a future refactor

**Pull request merged**